spanner/changestreams/src/main/java/com/example/spanner/changestreams/ChangeStreamSample.java

```diff
@@ -0,0 +1,255 @@
+
+    final Timestamp startTimestamp = Timestamp.now();
+    // end = start + 30 seconds.
+    final Timestamp endTimestamp = Timestamp.ofTimeSecondsAndNanos(
+        startTimestamp.getSeconds() + 30, startTimestamp.getNanos());
+    final String startTimestampStr = startTimestamp.toString();
+    final String endTimestampStr = endTimestamp.toString();
+
+    // Execute an initial query to get partition tokens.
+    System.out.println("Executing change stream initial query.");
+    List<ChangeStreamRecord> initialQueryRecords = executeChangeStreamQueryAndPrint(
+        dbClient, changeStreamName, startTimestampStr, endTimestampStr, null);
+
+    System.out.println("Executing change stream partition queries.");
+    for (ChangeStreamRecord record : initialQueryRecords) {
+      // Executes a partition query to print data records that we just inserted.
+      if (record instanceof ChildPartitionsRecord) {
+        for (ChildPartition childPartition :
+            ((ChildPartitionsRecord) record).getChildPartitions()) {
+          executeChangeStreamQueryAndPrint(
+              dbClient, changeStreamName, startTimestampStr, endTimestampStr,
+              childPartition.getToken());
+        }
+      }
+    }
+  }
+
+  // Insert two rows into Singers table.
+  static void insertRows(DatabaseClient client, String tableName) {
+    client.write(
+        ImmutableList.of(
+        Mutation.newInsertOrUpdateBuilder(tableName)
+          .set("SingerId").to(1)
+          .set("FirstName").to("Melissa")
+          .set("LastName").to("Garcia")
+          .build(),
+        Mutation.newInsertOrUpdateBuilder(tableName)
+          .set("SingerId").to(2)
+          .set("FirstName").to("Dylan")
+          .set("LastName").to("Shaw")
+          .build()
+      )
+    );
+  }
+
+  // Execute a change stream query, return and print out the result records.
+  public static List<ChangeStreamRecord> executeChangeStreamQueryAndPrint(
+      DatabaseClient dbClient, String changeStreamName, String startTimestamp,
+      String endTimestamp, String partitionToken) {
+    System.out.println("Executing a change stream query with: "
+        + "start_timestamp => " + startTimestamp
+        + ", end_timestamp => " + endTimestamp
+        + ", partition_token => " + partitionToken
+        + ", heartbeat_milliseconds => 5000"
+        + ", read_options => null");
+
+    final String query =
+        String.format("SELECT * FROM READ_%s ("
+          + "start_timestamp => @startTimestamp,"
+          + "end_timestamp => @endTimestamp,"
+          + "partition_token => @partitionToken,"
+          + "read_options => null,"
+          + "heartbeat_milliseconds => @heartbeatMillis"
+          + ")", changeStreamName);
+
+    final ResultSet resultSet =
+        dbClient
+          .singleUse()
+          .executeQuery(
+            Statement.newBuilder(query)
+              .bind("startTimestamp").to(Timestamp.parseTimestamp(startTimestamp))
```

Comment:
If you give this method a `Timestamp` instance instead of a `String` you would not need to parse it here.

Reply:
Done.